### ISSUE TITLE
Upgrade to web3py 5.13.1+

### ISSUE THREAD

**Issue by fubuloubu** [vyperlang/vyper#2259]
### What's your issue about?
web3py v5.13.1 introduced a breaking change that affected the error classes emitted during failed calls. (See https://github.com/ethereum/web3.py/pull/1809 for more information)

### How can it be fixed?
Upgrade to the result of whatever fix is applied for the issue mentioned above.

It is entirely in the test suite, specifically the `assert_tx_failed` fixture

**Comment by fubuloubu** [comment]
Waiting on https://github.com/ethereum/web3.py/pull/1813 to merge/release

**Comment by charles-cooper** [comment]
closing in favor of https://github.com/vyperlang/vyper/issues/2407
